drivers: pwm: pwm_sam: polarity support and SAM4S support

pdgendt opened the pull request:
This PR introduces (inverted) polarity support for the Atmel SAM PWM driver and adds the driver to the SAM4S SoC.

**Note:** The PWM driver was operating in inverted mode, so this changes the default behavior. But as `#pwm-cells` has been updated, users need to modify the device tree anyway.

Tested on a custom board.

pdgendt committed:
drivers: pwm: pwm_sam: Add alias definition for different SoCs

The PWM definitions for Atmel SAM SoCs can differ slightly.

This commit adds support for the PWM defines used by SAM4S.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
dts: arm: atmel: Add PWM support to Atmel SAM4S

Add device tree binding for PWM peripheral in Atmel SAM4S SoCs.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
drivers: pwm: pwm_sam: Add inverted polarity support

Set CPOL bit in Channel Mode Register if the PWM polarity is inverted.

Fixes #51544

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

nandojve commented on boards/arm/sam_v71_xult/sam_v71_xult-common.dtsi:
```diff
@@ -46,7 +46,7 @@
 		compatible = "pwm-leds";
 		pwm_led0: pwm_led_0 {
-			pwms = <&pwm0 0 PWM_MSEC(20)>;
+			pwms = <&pwm0 0 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
```
Look at [commit-guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines) about to automatically link a commit with an issue., since this should close #51544.

nandojve reviewed (changes requested):
Hi @pdgendt ,

Thank you for this contribution. This change require proper address on CI. To do that, it will be required another commit to enable feature on sam4s_xplained:
  * update features tables at doc/index.rst
  * create pins definitions at sam4s_xplained-pinctrl.dtsi
```diff
 &pinctrl {
+       pwm_default: pwm_default {
+               group1 {
+                       pinmux = <PA12B_PWM_PWMH1>,
+                                <PA13B_PWM_PWMH2>,
+                                <PA14B_PWM_PWMH3>;
+               };
+       };
+
```
  * add pwm-0 alias at sam4s_xplained.dts
  * enable pwm0 at sam4s_xplained.dts
```diff
+&pwm0 {
+       status = "okay";
+
+       pinctrl-0 = <&pwm_default>;
+       pinctrl-names = "default";
+};
+
```
    * add pwm entry at sam4s_xplained.yaml
```diff
   - memc
+  - pwm
   - spi
```

pdgendt committed:
boards: arm: sam4s_xplained: Enable PWM

Update SAM4S Xplained to support PWM. Add PWM entry to device tree and
pin control.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt commented:
> Thank you for this contribution. This change require proper address on CI. To do that, it will be required another commit to enable feature on sam4s_xplained

@nandojve thanks for the review, I've added the entry to the `sam4s_xplained` board, could you test & verify, please?

nandojve commented:
> > Thank you for this contribution. This change require proper address on CI. To do that, it will be required another commit to enable feature on sam4s_xplained
> 
> @nandojve thanks for the review, I've added the entry to the `sam4s_xplained` board, could you test & verify, please?

Yes, please, add missing `pwm-0 alias at sam4s_xplained.dts` https://github.com/zephyrproject-rtos/zephyr/blob/781ece7673b8d9ca49375635efa4be4382d465d4/boards/arm/sam_v71_xult/sam_v71_xult-common.dtsi#L18
